docs/calculations.mdx

```diff
@@ -0,0 +1,422 @@
+
+<PowerModelDefaults/>
+
+### Video player characteristics
+
+Based on actual data transfer of https://vjs.zencdn.net/8.10.0/video.min.js on February 23, 2024
+
+<VideoPlayerDefaults/>
+
+### Media size
+
+Based on [YouTube recommendations](https://support.google.com/youtube/answer/1722171), [Wikipedia](https://en.wikipedia.org/wiki/JPEG)
+
+<MediaSizeDefaults/>
+
+### Device size
+
+From common devices (Quad HD 27” monitor, iPhone 13, Nexus 5X, iPad Air 1/2, iPad 2/3, Nexus 9, 1080P TV)
+
+<DeviceSizeDefaults/>
+
+### Device energy use and embodied emissions
+
+See [Consumer Devices](./consumer_devices)
+
+<ConsumerDeviceDefaults/>
+
+### Time in view for non-video ads
+
+Observations from various channels
+
+<TimeInViewDefaults/>
+
+### Ad platform request sizes by channel
+
+Observations from various channels
+
+<AdPlatformDefaults/>
+
+## Lookups from external sources
+
+### Carbon intensity by country, region, and UTC Date/Time
+
+Providers should clearly and publicly document:
+- Which data provider is used for carbon intensity data (for instance, WattTime or ElectricityMaps)
+- What carbon intensity metric is used (marginal vs average)
+
+`lookup_carbon_intensity_gco2e_per_kwh(country, region, utc_datetime)` should look up the carbon intensity in gCO2e based on 100-year global warming potential (GWP100) from a database at no less than hourly granularity.
+
+## Calculating Emissions
+
+### Creative Delivery - Data Transfer
+
+```
+creative_data_transfer_emissions_gco2pm =
+      creative_data_transfer_usage_emissions_gco2pm +
+      creative_data_transfer_embodied_emissions_gco2pm
+```
+
+For all channels other than CTV/BVOD, use the conventional model:
+
+```
+creative_data_transfer_bytes =
+      video_bytes +
+      audio_bytes +
+      image_bytes +
+      ad_format.video_player.size_bytes +
+      ad_format.other_asset_bytes
+
+creative_data_transfer_usage_emissions_gco2pm =
+      creative_data_transfer_bytes / 1024 x
+      default_usage_kwh_per_kb[network_type] x
+      lookup_carbon_intensity_gco2e_per_kwh(country, region, utc_datetime)
+
+creative_data_transfer_embodied_emissions_gco2pm =
+      creative_data_transfer_bytes / 1024 x
+      default_network_embodied_emissions_gco2e_per_kb[network_type]
+```
+
+If data transfer is provided in the delivery row, use it:
+
+```
+image_bytes = creative_image_data_transfer_bytes
+video_bytes = creative_video_data_transfer_bytes + creative_video_vast_bytes
+audio_bytes = creative_audio_data_transfer_bytes
+```
+
+Otherwise, calculate data transfer:
+
+```
+image_bytes = SUM(image_sizes[n].width x image_sizes[n].height) x 3 / default_image_compression_ratio
+
+audio_duration = creative_audio_duration_seconds ?? ad_format.audio_duration_seconds
+audio_bytes = audio_duration_seconds x default_audio_bitrate[device_type]
+
+default_bitrate =
+      ad_format.is_primary_experience ?
```

Comment:
@bokelley If we don't have an ad format - should we default this based on the channel?

Reply:
I think in this context we would always have an ad format. From a spec perspective should we define the default ad format for each channel or is that an implementation choice?